src/DIRAC/TransformationSystem/Service/TransformationManagerHandler.py

```diff
@@ -33,6 +34,29 @@ def initializeHandler(cls, serviceInfoDict):
 
         return S_OK()
 
+    def checkPermissions(self, transName):
```

Comment:
@fstagni Is username really in v8.0 ?

Reply:
But if username is not in v8.0 then I can't use it ? In this case there would have to be two versions of the code on for 8.0 and one for 8.1 and up. Which is do-able.

Reply:
IIRC username is there in v8.0 too.

Reply:
I still think that this is a good candidate for being a decorator.

Reply:
@sfayer and me just had a look at decorators, and it looks like this is only going to work if all decorated functions have the transformation name as the second argument. Which at the moment they seem to do, but this feels a bit fragile.

Reply:
@fstagni @andresailer Given previous comments, any opinion on decorator worthiness ?

Reply:
It's your decision, if you feel it won't be better with it, leave it outside.

Reply:
At the same, seeing that you had to call the same `if not self.checkPermissions(transName)` in really a lot of methods, and that extensions might potentially override those methods, I am thinking that it might be better to use a decorator anyway. If you want to attach it to a large-ish number of methods in one go, and having this in extended classes too you can use a metaclass (or mayb the `inspect` module, see e.g. https://stackoverflow.com/questions/3467526/attaching-a-decorator-to-all-functions-within-a-class.)

Reply:
@fstagni We don't want it on *all* the functions through, quite a lot of them are read/summary/some internal workings that I don't understand and I'd rather not touch them.

Reply:
@andresailer Do you have an extension ? Does CTA ? In principle it should be backwards compatible.

Reply:
I don't have an extension of the TransformationManager